File: api_CBLA/src/main/java/com/daw/persistence/entities/Partido.java
package com.daw.persistence.entities;

import java.time.LocalDate;
import java.time.LocalTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "Partido")
@Getter
@Setter
@NoArgsConstructor
public class Partido {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

	@Column(name = "id_equip_local")
	private Integer idEquipLocal;

	@Column(name = "id_equip_visit")
	private Integer idEquipVisit;

	private LocalDate fecha;

	private LocalTime hora;

	@Column(name = "puntos_local", nullable = false, unique = true)
	private Integer puntosLocal;

	@Column(name = "puntos_visitante", nullable = false, unique = true)
	private Integer puntosVisitante;

	@Column(name = "id_ganador")
	private Integer idGanador;

	@Column(name = "id_jornada")
	private Integer idJornada;

	@ManyToOne
	@JoinColumn(name = "id_equip_local", referencedColumnName = "id", insertable = false, updatable = false)
	private Equipo equip_local;

	@ManyToOne
	@JoinColumn(name = "id_equip_visit", referencedColumnName = "id", insertable = false, updatable = false)
	private Equipo equip_visit;

	@ManyToOne
	@JoinColumn(name = "id_ganador", referencedColumnName = "id", insertable = false, updatable = false)
	private Equipo ganador;

	@ManyToOne
	@JoinColumn(name = "id_jornada", referencedColumnName = "id", insertable = false, updatable = false)
	private Jornada jornada;

}

File: front_BLA/src/app/models/partido.ts
import { Equipo } from "./equipo";
import { Jornada } from "./jornada";

export class Partido {
    private _id: number;
    private _equip_local: Equipo;
    private _equip_visit: Equipo;
    private _fecha: Date;
    private _hora: string;
    private _puntosLocal: number;
    private _puntosVisitante: number;
    private _ganador: Equipo;
    private _jornada: Jornada;

    constructor(id: number, equip_local: Equipo, equip_visit: Equipo,fecha: Date, hora: string, 
        puntosLocal: number, puntosVisitante: number, ganador: Equipo, jornada: Jornada) {
        this._id = id;
        this._equip_local = equip_local;
        this._equip_visit = equip_visit;
        this._fecha = fecha;
        this._hora = hora;
        this._puntosLocal = puntosLocal;
        this._puntosVisitante = puntosVisitante;
        this._ganador = ganador;
        this._jornada = jornada;
    }

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get equip_local(): Equipo {
        return this._equip_local;
    }

    public set equip_local(value: Equipo) {
        this._equip_local = value;
    }

    public get equip_visit(): Equipo {
        return this._equip_visit;
    }

    public set equip_visit(value: Equipo) {
        this._equip_visit = value;
    }

    public get fecha(): Date {
        return this._fecha;
    }

    public set fecha(value: Date) {
        this._fecha = value;
    }

    public get hora(): string {
        return this._hora;
    }

    public set hora(value: string) {
        this._hora = value;
    }

    public get puntosLocal(): number {
        return this._puntosLocal;
    }

    public set puntosLocal(value: number) {
        this._puntosLocal = value;
    }

    public get puntosVisitante(): number {
        return this._puntosVisitante;
    }

    public set puntosVisitante(value: number) {
        this._puntosVisitante = value;
    }

    public get ganador(): Equipo {
        return this._ganador;
    }

    public set ganador(value: Equipo) {
        this._ganador = value;
    }

    public get jornada(): Jornada {
        return this._jornada;
    }

    public set jornada(value: Jornada) {
        this._jornada = value;
    }
}